### PR TITLE
Fix incorrect assertion in builder.py

### PR DESCRIPTION
**Pull request by dreamgonfly**
Fix incorrect num_examples comparison assertion in builder.py

### CODE REVIEW

**Comment by lhoestq** [comment]
Hi ! The SplitInfo is not always available. By default you would get `split_info.num_examples == 0`
So unfortunately we can't use this assertion you suggested

**Comment by dreamgonfly** [comment]
> Hi ! The SplitInfo is not always available. By default you would get `split_info.num_examples == 0`
> So unfortunately we can't use this assertion you suggested

Then it would be better to just remove the assertion, because the existing assertion does nothing.

File: src/datasets/builder.py [hunk 1]
```diff
@@ -981,7 +981,7 @@ def _prepare_split(self, split_generator):
             finally:
                 num_examples, num_bytes = writer.finalize()
 
-        assert num_examples == num_examples, f"Expected to write {split_info.num_examples} but wrote {num_examples}"
+        assert split_info.num_examples == num_examples, f"Expected to write {split_info.num_examples} but wrote {num_examples}"
```
Comment (dreamgonfly) [inline src/datasets/builder.py:984]:
```suggestion

```

**Review by lhoestq — APPROVED** [review]
Thanks.
You can ignore the CI fail on the documentation, this has been fixed on master

**Pull request merged**